### DISCUSSION TITLE
Login page lets you unselect the only login type

### DISCUSSION THREAD

**Discussion by jtapsl** [requarks/wiki#3616] [Error / Bug Report]
**Describe the bug**
In Wiki.js version 2.5.144 , if LDAP authentication is enabled and Hide Local Authentication Provider is set then the login page sometimes gives the error "this.selectedStrategy is undefined"

**To Reproduce**
1. Enable and configure LDAP / Active Directory authentication.
2. In the Security section of the administration area, enable the option "Hide Local Authentication Provider".
3. Log out.
4. At the login screen under Select Authentication Provider click "LDAP / Active Directory". Note that clicking seems to toggle the provider on/off.
5. While the provider is toggled off, attempt to log in.
6. You receive an error message. In FireFox I get "this.selectedStrategy is undefined" and in Chrome I get "Cannot read property 'key' of undefined".

**Expected behavior**
Do not allow the only login provider to be unselected, or provide a more clear error message if the user does so.

**Screenshots**
![wikijs_loginerror](https://user-images.githubusercontent.com/42975697/94822576-cfd73300-03d0-11eb-8882-9d7e3b6ccee5.png)

**Host Info (please complete the following information):**
 - OS: RHEL 8.2
 - Wiki.js version: 2.5.144
 - Database engine: PostgreSQL 10.14

**Additional context**
N/A

**Comment by ag-gaphp** [comment]
I am running into this same issue. I setup Azure AD and told the login page to only show the Azure AD option. However, I **did not** disable local auth in the admin area, only hid it so you have to use `?all` in the URL to see the option. My admin account is the only local account, and all Azure AD accounts are read-only.

Once I enabled Azure AD and tested my login, I am completely unable to login using the local method. I **always** get the error `this.selectedStrategy is undefined` when I try to use local. I am completely locked out of my instance now and cannot login as admin.

Can I recover from this somehow, or do I have to nuke the entire thing and start over? This seems like a bug, I saw this was originally an issue then moved here, but there's no discussion surrounding it. Hoping someone can help and save me having to re-do all the work I've already put in to set this instance up.

**Reply by ag-gaphp** [reply]
I should clarify, I tested my Azure AD login in a Private Window. When I went back to edit some settings in the normal browser window, which was already logged in as the Admin, got auto-logged out and forced to the login page.

**Reply by ag-gaphp** [reply]
I don't know what happened differently, but after trying what feels like 100 attempts to login (more realistically ~25 attempts), one finally successfully logged in using local authentication. I used that opportunity to turn my Azure account into an admin, just in case this happens again. Still though, would be **excellent** to sort this out.
